File: sdk/fsharp/src/HiTechCloudSDK/EmailMfaV2.fs
namespace HiTechCloud.SDK

/// EmailMfaV2: email-based MFA management
type EmailMfaV2Resource(httpClient: HttpClient) =

    /// GET /api/email_mfa_v2/status/{user_type}/{user_id} - Get MFA status
    member this.GetMfaStatus() =
        async {
            return! httpClient.getAsync("/api/email_mfa_v2/status/{user_type}/{user_id}")
        }

    /// POST /api/email_mfa_v2/send - Send one-time code
    member this.SendCode() =
        async {
            return! httpClient.postAsync("/api/email_mfa_v2/send")
        }

    /// POST /api/email_mfa_v2/verify - Verify one-time code
    member this.VerifyCode() =
        async {
            return! httpClient.postAsync("/api/email_mfa_v2/verify")
        }

    /// GET /api/email_mfa_v2/list/{user_type}/{user_id} - List active codes
    member this.ListActiveCodes() =
        async {
            return! httpClient.getAsync("/api/email_mfa_v2/list/{user_type}/{user_id}")
        }

    /// POST /api/email_mfa_v2/revokeall - Revoke all active codes
    member this.RevokeAllCodes() =
        async {
            return! httpClient.postAsync("/api/email_mfa_v2/revokeall")
        }

    /// POST /api/email_mfa_v2/disable - Disable MFA entirely
    member this.DisableMfa() =
        async {
            return! httpClient.postAsync("/api/email_mfa_v2/disable")
        }

File: sdk/clojure/src/hitechcloud/sdk/email_mfa_v2.clj
(ns hitechcloud.sdk.email_mfa_v2
  (:require [hitechcloud.sdk.http-client :as http]))

;; EmailMfaV2: email-based MFA management

(defn get_mfa_status "GET /api/email_mfa_v2/status/{user_type}/{user_id} - Get MFA status" [client & [params]] (http-get client "/api/email_mfa_v2/status/{user_type}/{user_id}" params))

(defn send_code "POST /api/email_mfa_v2/send - Send one-time code" [client & [params]] (http-post client "/api/email_mfa_v2/send" params))

(defn verify_code "POST /api/email_mfa_v2/verify - Verify one-time code" [client & [params]] (http-post client "/api/email_mfa_v2/verify" params))

(defn list_active_codes "GET /api/email_mfa_v2/list/{user_type}/{user_id} - List active codes" [client & [params]] (http-get client "/api/email_mfa_v2/list/{user_type}/{user_id}" params))

(defn revoke_all_codes "POST /api/email_mfa_v2/revokeall - Revoke all active codes" [client & [params]] (http-post client "/api/email_mfa_v2/revokeall" params))

(defn disable_mfa "POST /api/email_mfa_v2/disable - Disable MFA entirely" [client & [params]] (http-post client "/api/email_mfa_v2/disable" params))

